pkg/cli/render_csv.go

```diff
@@ -259,26 +268,99 @@ func (p *Project) parseToCSV(project *reports.Project, data *csvData) {
 				csvRecord = append(csvRecord, p.DomainName, project.Name, pSrv.ServiceInfo.Type,
-					pSrvRes.ResourceInfo.Name, strconv.FormatUint(pSrvRes.Quota, 10),
-					strconv.FormatUint(pSrvRes.Usage, 10), string(pSrvRes.ResourceInfo.Unit),
+					pSrvRes.ResourceInfo.Name, val["quota"], val["usage"], unit,
 				)
 			case p.Opts.Long:
 				csvRecord = append(csvRecord, p.DomainID, p.DomainName, project.UUID, project.Name, pSrv.ServiceInfo.Area,
 					pSrv.ServiceInfo.Type, pSrvRes.ResourceInfo.Category, pSrvRes.ResourceInfo.Name,
-					strconv.FormatUint(pSrvRes.Quota, 10), strconv.FormatUint(pSrvRes.Usage, 10),
-					string(pSrvRes.ResourceInfo.Unit), time.Unix(pSrv.ScrapedAt, 0).UTC().Format(time.RFC3339),
+					val["quota"], val["usage"], unit, time.Unix(pSrv.ScrapedAt, 0).UTC().Format(time.RFC3339),
 				)
 			default:
 				csvRecord = append(csvRecord, p.DomainID, project.UUID, pSrv.ServiceInfo.Type,
-					pSrvRes.ResourceInfo.Name, strconv.FormatUint(pSrvRes.Quota, 10),
-					strconv.FormatUint(pSrvRes.Usage, 10), string(pSrvRes.ResourceInfo.Unit),
+					pSrvRes.ResourceInfo.Name, val["quota"], val["usage"], unit,
 				)
 			}
 
 			*data = append(*data, csvRecord)
 		}
 	}
 }
+
+type rawValues map[string]uint64
+type convertedValues map[string]string
+
+// unitExp is map of limes.Unit to x, such that base-2 exponential of x
+// is the number of bytes for some specific unit.
+var unitExp = map[limes.Unit]float64{
+	limes.UnitKibibytes: 10,
+	limes.UnitMebibytes: 20,
+	limes.UnitGibibytes: 30,
+	limes.UnitTebibytes: 40,
+	limes.UnitPebibytes: 50,
+	limes.UnitExbibytes: 60,
+}
+
+func humanReadable(convert bool, unit limes.Unit, rv rawValues) (string, convertedValues) {
+	cv := make(convertedValues, len(rv))
+
+	// 'usage' is used to determine conversion suitability because it is common
+	// in all three hierarchies and is the lowest value amongst its counterparts
+	if unit == limes.UnitNone || lenUint(rv["usage"]) < 4 {
+		convert = false
+	}
+	if !convert {
+		for i, v := range rv {
+			cv[i] = strconv.FormatUint(v, 10)
+		}
+		return string(unit), cv
+	}
+
+	oldExp := unitExp[unit]
+	usage := rv["usage"]
+
+	var diffInExp float64
+	// 2^60 bytes (exbibytes) is the maximum supported unit
+	for diffInExp = 10; diffInExp <= (60 - oldExp); diffInExp += 10 {
+		// take a copy of usage inside the loop
+		usage := usage
+
+		usage = usage / uint64(math.Exp2(diffInExp))
+
+		if lenUint(usage) < 4 {
+			break
+		}
+	}
+
+	// determine the new unit
+	var newUnit limes.Unit
+	for i, v := range unitExp {
+		if v == (oldExp + diffInExp) {
+			newUnit = i
+		}
+	}
+
+	// convert values to the new unit
+	for i, v := range rv {
+		v := float64(v)
+		v = v / math.Exp2(diffInExp)
+		// round to second decimal place
+		v = math.Round(v*100) / 100
+		cv[i] = strconv.FormatFloat(v, 'f', -1, 64)
+	}
+
+	return string(newUnit), cv
+}
+
+// lenUint is a helper function that returns
+// the length (digit count) of a uint64.
+func lenUint(val uint64) int {
```

Comment:
Since you're using this function only in expressions of the form `lenUint(foo) < 4`, it can be removed and the checks can be replaced with just `foo < 10000`.